packages/web3-validator/src/web3_validator.ts

```diff
@@ -98,7 +98,11 @@ export class Web3Validator {
 			]);
 		}
 
-		if (!this._validator.validate(jsonSchema, data)) {
+		const schemaKey = JSON.stringify(jsonSchema);
```

Comment:
What do you think about using the hash of the `jsonSchema` as the key? 
We can use a simple hashing function that we need to add like this for example:
```javascript
String.prototype.hashCode = function() {
  var hash = 0,
    i, chr;
  if (this.length === 0) return hash;
  for (i = 0; i < this.length; i++) {
    chr = this.charCodeAt(i);
    hash = ((hash << 5) - hash) + chr;
    hash |= 0; // Convert to 32bit integer
  }
  return hash;
}

```

And then the code will be:
```suggestion
		const schemaKey = JSON.stringify(jsonSchema).hashCode()
```

@jdevcs & @luu-alex your recommendation is required. Do you think we should use a simple hashing function as above (that is great for the task)? Or should we utilize `keccak256` from `ethereum-cryptography/keccak` (we would need to add it to the dependencies of web3-validator)? Or should we not using any hash for this?

Thanks all,

Reply:
Good suggestion, Thanks @Muhammad-Altabba . We can definitely use hash as key for adding or querying schema in ajv. Currently there is already dependency of eth-cryptography in validator package https://github.com/web3/web3.js/blob/fe037a09e9da7882c5a2e1707a23e778c3558a74/packages/web3-validator/package.json#L31 so I'll suggest to use some existing hash algo ( for avoiding collisions ) instead of simple one.